### PR TITLE
rest template

### PR DESCRIPTION
**Pull request by sudhakar-vinayaka**

### CODE REVIEW

File: src/main/java/io/sudhakar/student/config/JwtConfiguration.java [hunk 1]
```diff
@@ -52,5 +51,7 @@ public AuthenticationManager authenticationManagerBean() throws Exception {
     @Bean
     public PasswordEncoder passwordEncoder() {
         return new BCryptPasswordEncoder();
+
+//        return NoOpPasswordEncoder.getInstance();
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/config/JwtConfiguration.java]: remove commented code

File: src/main/java/io/sudhakar/student/controller/GenericQueryController.java [hunk 2]
```diff
@@ -21,7 +21,6 @@ public GenericQueryController(GenericService genericService) {
         this.genericService = genericService;
     }
 
-
     @GetMapping
     public ResponseEntity<StudentResponse<List<Student>>> getAllStudents(@RequestParam("queries") String queryName) {
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/controller/GenericQueryController.java]: @RequestParam("queries") change to 
query

File: src/main/java/io/sudhakar/student/controller/JwtController.java [hunk 3]
```diff
@@ -48,11 +47,12 @@ public JwtController(AuthenticationManager authenticationManager, JwtUserDetails
 
             final String jwt = jwtTokenUtil.generateToken(userDetails);
 
+            log.info("api = /authenticate, method = POST, result = SUCCESS");
             return ResponseEntity.ok(new AuthenticationResponse(jwt));
+
         } catch (BadCredentialsException e) {
+            log.info("api = /authenticate, method = POST, result = ERROR");
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/controller/JwtController.java]: log exception

File: src/main/java/io/sudhakar/student/controller/RestTemplateController.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+@RequestMapping("/employee")
+public class RestTemplateController {
+    private final RestTemplateServiceImpl restTemplateService;
+
+    public RestTemplateController(RestTemplateServiceImpl restTemplateService) {
+        this.restTemplateService = restTemplateService;
+    }
+
+    @GetMapping
+    public ResponseEntity<StudentResponse<?>> getAllEmployee(@RequestParam("urlName") String urlName) {
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/controller/RestTemplateController.java]: don't take this as requestParam 
hardcode it in service

File: src/main/java/io/sudhakar/student/controller/RestTemplateController.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        log.info("api = /employee, method = GET, status = IN_PROCESS");
+
+        ServiceResponse<?> serviceResponse = restTemplateService.getAllEmployee(urlName);
+
+        log.info("api = /employee, method = GET, status = SUCCESS");
+        return ResponseEntity.status(serviceResponse.getHttpStatus()).body(new StudentResponse<>(serviceResponse.getData()));
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<StudentResponse<?>> getEmployeeById(@RequestParam("urlName") String urlName, @PathVariable int id) {
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/controller/RestTemplateController.java]: same comment as above for all the methods

File: src/main/java/io/sudhakar/student/dto/EmployeeGetAll.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.*;
+
+import java.util.List;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+@Data
+public class EmployeeGetAll {
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/dto/EmployeeGetAll.java]: changeClass Name as EmployeeResponseGetAll

File: src/main/java/io/sudhakar/student/repository/RowMapperRepository.java [hunk 7]
```diff
@@ -2,6 +2,7 @@
 
 import io.sudhakar.student.repository.queryDetails.Queries;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.DataAccessException;
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/repository/RowMapperRepository.java]: why do we need this import

File: src/main/java/io/sudhakar/student/service/UserService.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    ServiceResponse<Void> addUser(UserDetails userDetails);
+
+    ServiceResponse<Void> updateUser(UserDetails userDetails,int id);
+
+    ServiceResponse<Void> deleteUser(int id);
+
+
+
+//    ServiceResponse<Void> addUser(User task);
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/service/UserService.java]: remove commented code

File: src/main/java/io/sudhakar/student/service/impl/JwtUserDetailsService.java [hunk 9]
```diff
@@ -3,24 +3,23 @@
 import io.sudhakar.student.entity.UserDetailsEntity;
 import io.sudhakar.student.dto.User;
 import io.sudhakar.student.repository.UserDetailsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/service/impl/JwtUserDetailsService.java]: why do we need this import

File: src/main/java/io/sudhakar/student/service/impl/RestTemplateServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+            log.error("method = getAll, status = ERROR");
+            serviceResponse.setHttpStatus(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return serviceResponse;
+    }
+
+    public ServiceResponse<EmployeeResponse> getEmployeeById(String urlName, int id) {
+        ServiceResponse<EmployeeResponse> serviceResponse = new ServiceResponse<>();
+        try {
+            log.error("method = get, status = IN_PROCESS");
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/service/impl/RestTemplateServiceImpl.java]: loggers are not changes

File: src/main/java/io/sudhakar/student/urldetails/UrlClass.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
+package io.sudhakar.student.urldetails;
+
+import lombok.Getter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@Getter
+public class UrlClass {
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/urldetails/UrlClass.java]: EmployeeEndPointDetails

**Review by abhishek-gowda12 — CHANGES_REQUESTED** [review]

File: src/main/java/io/sudhakar/student/controller/RestTemplateController.java [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+        log.info("api = /employee, method = PUT, status = IN_PROCESS");
+
+        ServiceResponse<Void> serviceResponse = restTemplateService.updateEmployee(employee,id);
+
+        log.info("api = /employee, method = PUT, status = SUCCESS");
+        return ResponseEntity.status(serviceResponse.getHttpStatus()).build();
+    }
+
+    @DeleteMapping("{id}")
+    public ResponseEntity<StudentResponse<Void>> deleteEmployee(@PathVariable int id) {
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/controller/RestTemplateController.java]: pathVariable should have name refer other controller

File: src/main/java/io/sudhakar/student/dto/Employee.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
+package io.sudhakar.student.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Employee {
```
Comment (abhishek-gowda12) [inline src/main/java/io/sudhakar/student/dto/Employee.java:9]: have you tested this class there's no data or getter setter

File: src/test/java/io/sudhakar/student/controller/GenericQueryControllerTest.java [hunk 14]
```diff
@@ -183,16 +175,16 @@ void updateStudentInternalServerError() {
...
     }
 
-  /*
-  Given deleteStudent(id,queryName)
-  Then  ServiceResponse
-  Scenario SUCCESS
-  */
+    /*
+    Given deleteStudent(id,queryName)
+    Then  ServiceResponse
```
Comment (abhishek-gowda12) [inline src/test/java/io/sudhakar/student/controller/GenericQueryControllerTest.java]: update this comments for all the methods

**Review by abhishek-gowda12 — CHANGES_REQUESTED** [review]

File: src/main/java/io/sudhakar/student/config/JwtConfiguration.java (hunk 1 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/config/JwtConfiguration.java]: removed commented code

File: src/main/java/io/sudhakar/student/controller/GenericQueryController.java (hunk 2 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/controller/GenericQueryController.java]: chage name to query

File: src/main/java/io/sudhakar/student/controller/JwtController.java (hunk 3 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/controller/JwtController.java]: logged those  errrors

File: src/main/java/io/sudhakar/student/controller/RestTemplateController.java (hunk 4 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/controller/RestTemplateController.java]: hard coded in RestTemplateServiceImpl class

File: src/main/java/io/sudhakar/student/controller/RestTemplateController.java (hunk 5 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/controller/RestTemplateController.java]: hard coded for all methods in RestTemplateServiceImpl class

File: src/main/java/io/sudhakar/student/dto/EmployeeGetAll.java (hunk 6 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/dto/EmployeeGetAll.java]: changed class name from EmployeeGetAll to EmployeeResponseGetAll

File: src/main/java/io/sudhakar/student/repository/RowMapperRepository.java (hunk 7 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/repository/RowMapperRepository.java]: removed un used imports

File: src/main/java/io/sudhakar/student/service/UserService.java (hunk 8 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/service/UserService.java]: removed all comented code

File: src/main/java/io/sudhakar/student/service/impl/JwtUserDetailsService.java (hunk 9 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/service/impl/JwtUserDetailsService.java]: removed all un necessary imports

File: src/main/java/io/sudhakar/student/urldetails/UrlClass.java (hunk 11 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/urldetails/UrlClass.java]: changed class name  to EmployeeEndPointDetails

File: src/main/java/io/sudhakar/student/dto/Employee.java (hunk 13 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/dto/Employee.java:9]: added getter setter

File: src/test/java/io/sudhakar/student/controller/GenericQueryControllerTest.java (hunk 14 above)
Comment (sudhakar-vinayaka) [reply src/test/java/io/sudhakar/student/controller/GenericQueryControllerTest.java]: removed query name in comments for all method

File: src/main/java/io/sudhakar/student/controller/RestTemplateController.java (hunk 12 above)
Comment (sudhakar-vinayaka) [reply src/main/java/io/sudhakar/student/controller/RestTemplateController.java]: changed @pathvariable to @pathvariable("id")

File: src/test/java/io/sudhakar/student/controller/StudentControllerTest.java [hunk 15]
```diff
@@ -386,7 +386,7 @@ void updateStudentInternalServerError() {
         when(studentService.updateStudent(1, student))
                 .thenReturn(serviceResponse);
 
-        ResponseEntity<?> responseEntity = studentController.updateStudent(1, student);
+        ResponseEntity<Void> responseEntity = studentController.updateStudent(1, student);
         assertEquals(500, responseEntity.getStatusCodeValue());
 
     }
```
Comment (abhishek-gowda12) [inline src/test/java/io/sudhakar/student/controller/StudentControllerTest.java:392]: ResponseEntity<Void> change in comments something like this

**Review by abhishek-gowda12 — CHANGES_REQUESTED** [review]

File: src/test/java/io/sudhakar/student/controller/StudentControllerTest.java (hunk 15 above)
Comment (sudhakar-vinayaka) [reply src/test/java/io/sudhakar/student/controller/StudentControllerTest.java:392]: comments updated

**Review by abhishek-gowda12 — APPROVED** [review]

**Pull request merged**